xonsh/tools.py

```diff
@@ -1582,9 +1582,13 @@ def expanduser_abs_path(inp):
 def expand_case_matching(s):
```

Comment:
i know this comes a little late but i always feel like regular expressions is the best tool for parsing jobs like this, with regex the implementation is much cleaner and more efficient too! If you want we can try out implementing this with regex.